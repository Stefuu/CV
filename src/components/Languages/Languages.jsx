import React from 'react'
import PropTypes from 'prop-types'
import Flex from '../elements/Flex'
import Box from '../elements/Box'

const LanguageItem = (props) => (
  <Box mb={5}>
    <Box>
      <strong>{props.name}</strong><span> - {props.level}</span>
    </Box>
  </Box>
)

LanguageItem.propTypes = {
  name: PropTypes.string,
  level: PropTypes.string
}

const Languages = (props) => (
  <Flex
    mb={30}
    width='100%'
    flexDirection='column'>
    <Box
      mb={5}
      fontSize={24}
      color='#cc7000'>
      <strong>Idiomas:</strong>
    </Box>
    {
      props.items.map((item, index) => (
        <LanguageItem key={index} {...item} />
      ))
    }
  </Flex>
)

Languages.propTypes = {
  items: PropTypes.array
}

export default Languages
