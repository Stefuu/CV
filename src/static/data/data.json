{
	"header": {
		"name": "Stéfano Damiano",
		"description": "Desenvolvedor Front-End",
		"items": [
			{
				"name": "Phone",
				"text": "11 97027-9708"
			},
			{
				"name": "Nasc",
				"text": "08/08/1989"
			},
			{
				"name": "Cidade",
				"text": "São Paulo - SP"
            },
            {
				"name": "Email",
				"text": "mineiro.guaxupe@gmail.com"
			}
		]
    },
    "xperience": {
        "items": [
            {
                "company": "- TOVS (idexo)",
                "duration": "De: Jan/2018 até Hoje",
                "position": "Cargo: Desenvolvedor Front-End Sênior",
                "description": "Descrição: Criação e manutenção de um e-commerce capaz de se adequar a regras de negócio complexas de produtos da empresa",
                "highlights": [                    
                    "- Ambiente de inovação com intenção de renovar a cultura da velha TOTVS. Trabalhamos lado a lado com start-ups incubadas pela TOTVS, o que permite agregar muito sobre tecnologias e soluções muito quentes",
                    "- Tivemos a oportunidade de começar tudo do zero, portanto escolhemos a stack, que é: React com styled-components, styled-system, fazendo uso de context API, Graphql com Relay modern, Node (express) e MQTT (para mensagem entre micro-serviços)",
                    "- Criação de uma arquitetura de execução de regras de negócio no front-end que não são fixas no mesmo. Usamos uma estrutura declarativa com https://github.com/jwadhams/json-logic-js que permite que as regras sejam geradas pelo back-end e apénas consumidas e executadas pelo front (evitando um oceano de condições no front)."
                ]
            },
            {
                "company": "- Universo Online (UOL)",
                "duration": "De: Jul/2012 até Dez/2017",
                "position": "Cargo: Desenvolvedor Front-End Pleno",
                "description": "Descrição: Criação e manutenção de páginas do portal (layout, integrações com back-end e sistemas externos), criação e manutenção de sistemas internos (principalmente em node), criação de serviços",
                "highlights": [                    
                    "- Passei por diversas areas de desenvolvimento da empresa, ganhando bom conhecimento das dificuldades e desafios de colocar e manter um portal com milhões de acessos em produção.",
                    "- Passei pelas dificuldades de se criar layouts responsivos cross-browser que precisam rodar em navegadores antigos.",
                    "- Desenvolvi integrações com ferramentas experimentais do google e facebook. Como o UOL tinha parceria com essas empresas, e era um bom laboratório pela quantidade de acessos, eles liberavam algumas ferramentas novas antes do lançamento. Ferramentas como AMP, Instant Articles e Facebook Messenger BOTS.",
                    "- Aprendi muito sobre arquitetura de back-end trabalhando na criação do micro-serviço do Facebook Messenger BOT, pelo fato de ser necessário fazer envio de milhões de mensagens (quando aparecia alguma notícia bombástica), aprendi muito sobre como escalar uma aplicação node, kubernetes, sobre uso de filas (rabbitmq principalmente) e banco de dados não relacional (mongodb)"
                ]
            },
            {
                "company": "- KAITS - Sistema educacional",
                "duration": "De: Fev/2011 até Jan/2012",
                "position": "Cargo: Estagiário",
                "description": "Descrição: Atuação com javascript vanilla na construção e manutenção de um sistema web para escolas de idioma",
                "highlights": [                    
                    "-Uso de javascript sem nenhum framework foi uma experiência interessante, principalmente por estar começando a usar js na época"                      
                ]
            }
        ]
    },
    "skills": [
        "Javascript",
        "ES20XX",
        "HTML5",
        "CSS3",
        "React",
        "Graphql",
        "Relay",
        "Unity Test",
        "Jest",
        "Enzyme",
        "Nightwatch",
        "Linux",
        "GIT",
        "Docker",
        "Gulp / Grunt",
        "Node",
        "Mongodb",
        "Rabbitmq",
        "SCRUM"
    ],
    "education": {
        "items": [
            {
                "name": "Universidade Presbiteriana Mackenzie",
                "type": "Bacharel em Ciência da Computação",
                "duration": "2008 - 2012"
            }
        ]
    },
    "languages": {
        "items": [
            {
                "name": "Português",
                "level": "Nativo"
            },
            {
                "name": "Inglês",
                "level": "Intermediário"
            },
            {
                "name": "Espanhol",
                "level": "Básico"
            }
        ]
    }
}
