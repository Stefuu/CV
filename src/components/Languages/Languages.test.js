import React from 'react'
import { render } from 'react-testing-library'
import Languages from './Languages'
import data from '../../static/data/data.json'

describe('Languages', () => {
  it('Deve renderizar [snapshot]', async () => {
    const { container } = render(<Languages {...data.languages} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
