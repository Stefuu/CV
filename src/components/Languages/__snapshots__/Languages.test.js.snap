// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Languages Deve renderizar [snapshot] 1`] = `
<div
  class="sc-bdVaJa bGLETG"
>
  <div
    class="sc-bdVaJa hJJydj"
  >
    <strong>
      Idiomas:
    </strong>
  </div>
  <div
    class="sc-bdVaJa cLzzJK"
  >
    <div
      class="sc-bdVaJa hvxHyA"
    >
      <strong>
        Português
      </strong>
      <span>
         - 
        Nativo
      </span>
    </div>
  </div>
  <div
    class="sc-bdVaJa cLzzJK"
  >
    <div
      class="sc-bdVaJa hvxHyA"
    >
      <strong>
        Inglês
      </strong>
      <span>
         - 
        Intermediário
      </span>
    </div>
  </div>
  <div
    class="sc-bdVaJa cLzzJK"
  >
    <div
      class="sc-bdVaJa hvxHyA"
    >
      <strong>
        Espanhol
      </strong>
      <span>
         - 
        Básico
      </span>
    </div>
  </div>
</div>
`;
